plugins/early_talker.js

```diff
@@ -41,6 +47,12 @@ exports.early_talker = function(next, connection) {
         return next();
     }
 
+    // Don't delay whitelisted IPs
+    if (plugin.whitelist && plugin.ip_in_list(plugin.whitelist, connection.remote_ip)) { // check connecting IP
```

Comment:
You can also move the `if (plugin.whitelist)` test into plugin.ip_in_list, simplifying the request dispatch.